tests/Http/Offer/AddVideoRequestHandlerTest.php

```diff
@@ -0,0 +1,149 @@
+            ->willReturn($videoId);
+
+        $this->addVideoRequestHandler->handle($addVideoRequest);
+
+        $this->assertEquals(
+            [
+                new AddVideo(
+                    new UUID('609a8214-51c9-48c0-903f-840a4f38852f'),
+                    (new Video(
+                        new UUID($videoId->toString()),
+                        new Url('https://www.youtube.com/watch?v=sdsd234')
+                    ))->withCopyrightHolder(new CopyrightHolder('publiq'))
+                ),
+            ],
+            $this->commandBus->getRecordedCommands()
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_adding_a_video_without_copyright_holder(): void
+    {
+        $addVideoRequest = $this->psr7RequestBuilder
+            ->withRouteParameter('offerId', '609a8214-51c9-48c0-903f-840a4f38852f')
+            ->withBodyFromString('{"url":"https://www.youtube.com/watch?v=sdsd234"}')
+            ->build('POST');
+
+        $videoId = \Ramsey\Uuid\Uuid::uuid4();
+        $this->uuidFactory->expects($this->once())
+            ->method('uuid4')
+            ->willReturn($videoId);
+
+        $this->addVideoRequestHandler->handle($addVideoRequest);
+
+        $this->assertEquals(
+            [
+                new AddVideo(
+                    new UUID('609a8214-51c9-48c0-903f-840a4f38852f'),
+                    new Video(new UUID($videoId->toString()), new Url('https://www.youtube.com/watch?v=sdsd234'))
+                ),
+            ],
+            $this->commandBus->getRecordedCommands()
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function it_requires_a_url(): void
+    {
+        $addVideoRequest = $this->psr7RequestBuilder
+            ->withRouteParameter('offerId', '609a8214-51c9-48c0-903f-840a4f38852f')
+            ->withBodyFromString('{"copyrightHolder":"publiq"}')
+            ->build('POST');
+
+        $this->assertCallableThrowsApiProblem(
+            ApiProblem::bodyInvalidData(
+                new SchemaError('/', 'The required properties (url) are missing')
+            ),
+            fn () => $this->addVideoRequestHandler->handle($addVideoRequest)
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function it_only_allows_supported_video_platforms(): void
+    {
+        $addVideoRequest = $this->psr7RequestBuilder
+            ->withRouteParameter('offerId', '609a8214-51c9-48c0-903f-840a4f38852f')
+            ->withBodyFromString('{"url":"https://www.google.com/?v=sdsd234"}')
+            ->build('POST');
+
+        $this->assertCallableThrowsApiProblem(
+            ApiProblem::bodyInvalidData(
+                new SchemaError(
+                    '/url',
+                    'The string should match pattern: ^http(s?):\/\/(www\.)?((youtube\.com\/watch\?v=([^\/#&?]*))|(vimeo\.com\/([^\/#&?]*)))'
+                )
+            ),
+            fn () => $this->addVideoRequestHandler->handle($addVideoRequest)
+        );
+    }
```

Comment:
Nit: A test that checks that `copyrightHolder` is a string and not for example an object with language keys could be useful as well

Reply:
I added an extra test with `copyrightHolder` value `123`